adapter/pkg/messaging/azure_connection.go

```diff
@@ -37,45 +50,180 @@ func init() {
+				retryAttemptMessage = ""
+				if reconnectRetryCount > 0 {
+					retryAttemptMessage = "Retry attempt : " + strconv.Itoa(reconnectRetryCount)
+				}
+				logger.LoggerMgw.Errorf("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] %v " + retryAttemptMessage +
+					" Retrying after %s", processError, reconnectInterval)
+				time.Sleep(reconnectInterval)
+				continue
+			}
+			topicCreationSuccessful = true
+			subscriptionMetaDataList, processError = validateAndGetSubscriptionMetaDataList(subscriptionMetaDataList,
+				namespace, reconnectRetryCount, reconnectInterval, componentName,
+					servicebus.SubscriptionWithAutoDeleteOnIdle(&subscriptionIdleTimeDuration))
+			if processError != nil {
+				retryAttemptMessage = ""
+				if reconnectRetryCount > 0 {
+					retryAttemptMessage = "Retry attempt : " + strconv.Itoa(reconnectRetryCount)
+				}
+				logger.LoggerMgw.Errorf("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] %v. " + retryAttemptMessage + " " +
+					" Retrying after %s ", processError, reconnectInterval)
+				time.Sleep(reconnectInterval)
+				continue
+			}
+			return subscriptionMetaDataList, processError
+		}
+		if processError != nil {
+			logger.LoggerMgw.Errorf("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] %v. Retry attempted %d times. ",
+				processError, reconnectRetryCount)
+			return subscriptionMetaDataList, processError
 		}
-		logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Successfully received topic list ")
 	} else {
-		logger.LoggerMgw.Error("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Error occurred while trying get the namespace " +
-			"in azure service bus using the connection url " + eventListeningEndpoint + ":%v", err)
+		//Any error which comes to this point is because the connection url is not up to the expected format
+		//Hence not retrying
+		logger.LoggerMgw.Errorf("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Error occurred while trying get the namespace "+
+			"in azure service bus using the connection url %s :%v", eventListeningEndpoint, processError)
 	}
-	return namespace, availableTopics, err
+	return subscriptionMetaDataList, processError
 }
 
 // InitiateConsumers to pass event consumption
-func InitiateConsumers(ns *servicebus.Namespace, availableTopicList []*servicebus.TopicEntity, componentName string,
-	subscriptionIdleTimeDuration time.Duration) {
-	bindingKeys := []string {tokenRevocation, notification}
+func InitiateConsumers(subscriptionMetaDataList []Subscription, reconnectInterval time.Duration) {
+	for _, subscriptionMetaData := range subscriptionMetaDataList {
+		go func(subscriptionMetaData Subscription) {
+			logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] starting the consumer for topic : " +
+				subscriptionMetaData.topicName)
+			startBrokerConsumer(subscriptionMetaData, reconnectInterval)
+			select {}
+		}(subscriptionMetaData)
+	}
+}
 
+func validateAndCreateTopicForSubscription(availableTopicList []*servicebus.TopicEntity, ns *servicebus.Namespace) error {
```

Comment:
can we simply say createTopicIfNotExists

Reply:
+1